Automattic/wp-calypso
# Lint: use @wordpress/stylelint-config preset

#### Proposed Changes

This PR updates `.stylelintrc` to use [`@wordpress/stylelint-config`](https://www.npmjs.com/package/@wordpress/stylelint-config) preset and removes certain rules that are already provided by `@wordpress/stylelint-config`.

#### Testing Instructions

It is assumed that you use VS Code for testing

* Ensure that you have the official [Stylelint](https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint) extension installed and active.
* Checkout this PR
* Ensure that you have these settings in `settings.json` (`Cmd + Shift + P` > `Preferences: Open Settings (JSON)`)
```jsonc
	"stylelint.validate": ["css", "scss", "sass"],
	"[css][scss][sass]": {
		"editor.formatOnSave": false,
		"editor.defaultFormatter": "stylelint.vscode-stylelint"
	},
	"editor.codeActionsOnSave": {
		"source.fixAll.stylelint": true
		// other actions
	},
```
* Reload VS Code: `Cmd + Shift + P` > `Developer: Reload Window`
* Open a SCSS file e.g. `client/me/security-2fa-initial-setup/style.scss`
* Mess up the formatting e.g. add empty lines, use spaces for indentation, etc.
* Open problems view (`Cmd + Shift + P` > `Problems: Focus on Problems View`)
* Confirm that you see the problems reported by Stylelint.
* Save Changes (`Cmd + S`)
* Confirm that the file is formatted as expected.

#### Pre-merge Checklist






Related to #39568


<img width="585" alt="Screenshot 2022-08-24 at 4 40 22 PM" src="https://user-images.githubusercontent.com/18226415/186404280-bc37956c-ae5d-4628-b7b0-1d205a14ed00.png">
